### DISCUSSION TITLE
How to add Event Listener dynamically?

### DISCUSSION THREAD

**Discussion by dvorakchen** [leptos-rs/leptos#2423] [Q&A] [answered]
I was try to add a event listener to body, and I found a way:

```rust
create_effect(|_| {
        document().body().unwrap().add_event_listener_with_callback(
            "click",
            &Function::new_no_args(
                "
        JS code here
        ",
            ),
        )
    });
```

but as you see I can only write JS code in callback function, that mean I cannot use signal.

is anybody has better idea?

**Comment by gbj** [comment, marked as answer]
To just add an event listener on the window, use [one of these helpers](https://docs.rs/leptos/latest/leptos/?search=Window_event_listener)

If you need something more specific than that, you should drop into using `wasm-bindgen`'s `Closure` type to wrap Rust code as a JS function, rather than creating a function from a JS string. See their docs [here](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/closure/struct.Closure.html)

**Reply by caligin** [reply]
:information_source: wasm-bindgen docs moved, [here's the doc page for `Closure`](https://docs.rs/wasm-bindgen/latest/wasm_bindgen/closure/struct.Closure.html) on docs.rs
